### PR TITLE
fix: filelock dependency not required

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/hubble/helper.py [hunk 1]
```diff
@@ -287,10 +282,18 @@ def wrapper(*args, **kwargs):
...
+                help_text=f'FileLock is needed to guarantee non-concurrent access to the'
+                f'cache_file {cache_file}',
+            ):
+                import filelock
+
+                file_lock = filelock.FileLock(f'{cache_file}.lock', timeout=-1)
 
             cache_db = None
-            with filelock.FileLock(f'{cache_file}.lock', timeout=-1):
+            with file_lock:
```
Comment (cristianmtr) [inline jina/hubble/helper.py:296]: Should we even allow a nullcontext? Maybe make required=True?

Comment (JoanFM) [reply jina/hubble/helper.py:296]: required True means an exception will be raised and failed.

Comment (cristianmtr) [reply jina/hubble/helper.py:296]: Yes, that was my point. Shouldn't make this required? What happens if you don't have the actual package and just have a nullcontext?

Comment (JoanFM) [reply jina/hubble/helper.py:296]: in a very rare case where Flows are run in multiprocess or multithreaded way

**Pull request merged**